src/vm/runners/builtin_runner/range_check.rs

```diff
@@ -21,23 +29,31 @@ pub struct RangeCheckBuiltinRunner {
     pub(crate) cells_per_instance: u32,
     pub(crate) n_input_cells: u32,
     inner_rc_bound: usize,
-    pub _bound: BigInt,
+    pub _bound: Option<Felt>,
     pub(crate) _included: bool,
     n_parts: u32,
     instances_per_component: u32,
 }
 
 impl RangeCheckBuiltinRunner {
     pub fn new(ratio: u32, n_parts: u32, included: bool) -> RangeCheckBuiltinRunner {
-        let inner_rc_bound = 1usize << 16;
+        let inner_rc_bound = 1_usize << 16;
+
+        let bound = Felt::one().shl(16 * n_parts);
+        let _bound = if n_parts != 0 && bound.is_zero() {
```

Comment:
The only way for `bound.is_zero()` to be true is if `n_parts == 0`. For `CAIRO_PRIME != 2`, no power of `2` can be evenly divided by `CAIRO_PRIME`, so converting it to a `Felt` will never result in `0`.
On that note:
1. Check it before creating `bound`, so you can avoid creating it in the first place.
2. Don't create it twice, instead of calling `Felt::new(bound)` just use `bound`.